planner/core/logical_plans_test.go

```diff
@@ -179,9 +170,9 @@ func (s *testUnitTestSuit) TestIndexPathSplitCorColCond(c *C) {
 	}
 	collate.SetNewCollationEnabledForTest(true)
 	for _, tt := range testCases {
-		comment := Commentf("failed at case:\nexpr: %v\ncorColIDs: %v\nidxColIDs: %v\nidxColLens: %v\naccess: %v\nremained: %v\n", tt.expr, tt.corColIDs, tt.idxColIDs, tt.idxColLens, tt.access, tt.remained)
-		filters, err := expression.ParseSimpleExprsWithNames(s.ctx, tt.expr, totalSchema, names)
-		c.Assert(err, IsNil, comment)
+		comment := fmt.Sprintf("failed at case:\nexpr: %v\ncorColIDs: %v\nidxColIDs: %v\nidxColLens: %v\naccess: %v\nremained: %v\n", tt.expr, tt.corColIDs, tt.idxColIDs, tt.idxColLens, tt.access, tt.remained)
+		filters, err := expression.ParseSimpleExprsWithNames(ctx, tt.expr, totalSchema, names)
+		require.NoError(t, err, comment)
```

Comment:
```
require.NoErrorf(t, err, comment)
```

ditto other.

Reply:
I think for this line we can have different ways, like
1. `require.NoErrorf(t, err, comment)`
or
2.
```
comment := "failed at case:\nexpr: %v\ncorColIDs: %v\nidxColIDs: %v\nidxColLens: %v\naccess: %v\nremained: %v\n"
require.NoErrorf(t, err, comment, tt.expr, tt.corColIDs, tt.idxColIDs, tt.idxColLens, tt.access, tt.remained)
```
3. `require.NoError(t, err, comment)`

I think comment is already a well-formatted string, so `NoErrorf` doesn't really help much here. It will call `assert.NoErrorf` and then `NoError` eventually. So (1) is doing some extra work. With (2) we can utilize the benefit from `NoErrorf`, but i think it's no necessary.